Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -33,17 +39,33 @@ TextEditorWidget::TextEditorWidget()
     });
 
     auto open_action = GAction::create("Open document", { Mod_Ctrl, Key_O }, GraphicsBitmap::load_from_file("/res/icons/16x16/open.png"), [this](const GAction&) {
-        GFilePicker picker;
-
-        if (picker.exec() == GDialog::ExecOK) {
-            m_path = picker.selected_file().string();
+        Optional<String> open_name = GFilePicker::get_open_filepath();
+        if (open_name.has_value() && !open_name.value().is_null()) {
+            m_path = open_name.value();
             open_sesame(m_path);
         }
     });
 
     auto save_action = GAction::create("Save document", { Mod_Ctrl, Key_S }, GraphicsBitmap::load_from_file("/res/icons/16x16/save.png"), [this](const GAction&) {
-        dbgprintf("Writing document to '%s'\n", m_path.characters());
-        m_editor->write_to_file(m_path);
+        if (!m_path.is_empty()) {
+            if (!m_editor->write_to_file(m_path)) //Prettry redundant code here. not sure how I wanna approach it.
+                GMessageBox::show("Unable to save file.\n", "Error", GMessageBox::Type::Error, window());
+        } else {
+            Optional<String> open_name = GFilePicker::get_save_filepath();
+            if (open_name.has_value() && !open_name.value().is_null()) {
+                String path = open_name.value();
+                if (!file_exists(path)) {
```

Comment:
This kind of logic belongs in `GFilePicker`. If you are trying to overwrite an existing file, the file picker should ask you if you really want that, and by the time we return from `GFilePicker` we know we want to write to the file.
Otherwise we'll end up implementing this logic every time `GFilePicker` is used, which could be avoided :)